### ISSUE TITLE
Dynamically imported modules cannot be found in the asar 

### ISSUE THREAD

**Issue by cawa-93** [electron/electron#38957]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

26.0.0-alpha.8

### What operating system are you using?

Windows

### Operating System Version

Windows 11

### What arch are you using?

x64

### Expected Behavior

You can `import()` module in app inside asar.

### Actual Behavior

Attemt to `import()` some pachage inside `asar` leads to error

```js
import {app} from 'electron'; // ok ✅
import * as lodash from 'lodash'; // ok ✅

app
  .whenReady()
  .then(async () => {
    console.log('resolve', require.resolve('lodash')); // ok ✅
    console.log('cjs', require('lodash')); // ok ✅
    console.log('lodash', lodash); // ok ✅
    console.log('mjs', await import('lodash')); // fail. ERR_MODULE_NOT_FOUND ❌
    app.exit(0);
  })
```

Dynamic `import()` works as expected if app isn't archived inside asar.

### Additional Information

Reproduction repository - https://github.com/cawa-93/auto-update-issue-test

**Comment by YuhApps** [comment]
It happens on Linux and Windows with Electron 25 too. A workaround is put all node modules that require dynamic import and their dependencies in asar.unpacked folder. For example, assuming your main.js is located at the root folder of the project:

In debug
`const module = await import('module_name')`

In production
`const module = await import('../app.asar.unpacked/node_modules/module_name/index.js')`

However that's highly unpredictable since a module can depend on various modules. If a module in asar.unpacked folder requires a module inside the asar, the app will fail to start.

**Comment by cawa-93** [comment]
Seems fixed in current [29.0.0-nightly.20231027](https://www.npmjs.com/package/electron-nightly/v/29.0.0-nightly.20231027)
